### DISCUSSION TITLE
Move big js dropdown function out of html

### DISCUSSION THREAD

**Discussion by howdu** [joshhanley/livewire-autocomplete#17] [Help] [answered]
Without overwriting this whole file how can I move the big dropdown function into my app.js file?

https://github.com/joshhanley/livewire-autocomplete/blob/main/resources/views/autocomplete.blade.php

**Comment by joshhanley** [comment]
@howdu currently there's no easy way to do that. But I'd love to come up with a solution that allows it to be disabled in the blade file and instead be able to be included in a compiled JS file like `app.js`. Maybe based on a config option or something.

**Reply by howdu** [reply]
Cheers I had a play [doing that here](https://github.com/howdu/livewire-autocomplete/commit/5ba8fb45eba1070c3a5ee396e86c7a9a849c45b1) but kept getting "autocomplete is not defined". 

Moved the function into my app.js file and set it global with `window.autocomplete = function(config)` but Vite js module bundling prevents this from loading first.

**Reply by joshhanley** [reply, marked as answer]
@howdu I was experimenting with a rewrite on the weekend that actually includes an external `.js` file using a `script` tag, that can be turned off using a config option. This would in theory allow it to be included using a bundler or manually in a layout file. Still experimenting with it though.
